### ISSUE TITLE
Duplicated (even triplicated) Notifier interface

### ISSUE THREAD

**Issue by nikoliazekter** [UnderwaterApps/overlap2d#292]
`SimpleMediator` implements `Notifier` interface thrice! (kinda) https://github.com/UnderwaterApps/overlap2d/edit/master/overlap2d-common-api/src/com/puremvc/patterns/mediator/SimpleMediator.java This code is really weird. First, since `Mediator` already extends `Notifier`why do we need it to implement `Notifier` again? Second, why doesn't `BaseNotifier` implement `Notifier`? Third, `BaseNotifier` is `Notifier` and `Mediator` is `Notifier`. I think that it's better to remove `extends Notifier` from `Mediator` to make code clearer.  I actually was going to create PR with some changes but figured out that I need your agreement what do with all this code.

**Comment by azakhary** [comment]
Oh, this part is spooky, it was done by @saqsun , but I think it was also copied from somewhere. @saqsun can you help us out with advice on this? Thanks.

**Comment by nikoliazekter** [comment]
Apparently @saqsun can't help us with this.

**Comment by azakhary** [comment]
yeah. I guess I'll take a look. but this is copied from somewhere, so had to be correct. I'll check
